imports/client/components/PuzzleListPage.tsx

```diff
@@ -368,10 +358,11 @@ const PuzzleListView = ({
           <ViewControlsSection>
             <FormLabel>View puzzles by:</FormLabel>
             <ButtonToolbar className="puzzle-view-buttons">
-              <ToggleButtonGroup type="radio" className="mr-2" name="puzzle-view" defaultValue="group" value={displayMode} onChange={switchView}>
+              <ToggleButtonGroup type="radio" className="mr-2" name="puzzle-view" defaultValue="group" value={displayMode} onChange={setDisplayMode}>
                 <ToggleButton variant="outline-info" value="group">Group</ToggleButton>
                 <ToggleButton variant="outline-info" value="unlock">Unlock</ToggleButton>
               </ToggleButtonGroup>
+              <Button variant="info" className="mr-2" onClick={expandAllGroups}>Expand all</Button>
```

Comment:
It's sorta weird that we show this when the display mode is "unlock" since it literally does not impact that view at all?

Maybe we could show it only if the current displayMode is "group" (and maybe put it after `showSolved`)?

Reply:
Hmm. It feels a little jumpy to me to have it show/vanish:

https://user-images.githubusercontent.com/28167/155198194-3955e8a7-8725-429b-bb72-d93d6ea999d2.mov

Thoughts on disabling it instead?

https://user-images.githubusercontent.com/28167/155198370-97a8cb88-b2d0-40f6-94c6-b9108ab6fc8e.mov

(No objections to moving it to a different spot; I don't have a strong feeling)

Reply:
I think we're getting a lot of saturated-color buttons in that row which are all trying to draw the user's attention, and I wish we could make them feel a bit lighter or make them take up less space, since they're really not the most important content on the page, but alas.

I tried putting the "expand all" button on the far right (after the operator button set) and there were things I liked about it (weaker blob of a bunch of saturated-color buttons all together, and the churn seemed pretty minimal when removing the expand button when toggling view mode) but there were also things I disliked about it (it makes the wrapping on mobile super goofy, and then visually it feels like the Expand all button is logically a part of the filter search bar, which also doesn't feel *quite* right ontologically).

I'm not really sure what to do with positioning here.  Nothing we've tried yet feels *right*, really.  Part of me wonders if it's something we should make "expand all" (and maybe collapse all?) into a button that appears on hover (or on touchscreen devices: active focus/click) at the far right of the collapsible rows instead -- it'd keep the feature closer to what people are already going to reach for to interact with the thing by default anyway?  Hmmm, that might be a little weird since maybe it would feel contextually like the button only affects the children of the group that you're clicking on.

I dunno.  I can live with the current button location and keeping it there disabled when the display mode means that the button won't do anything, but I will ask that we also disable the button if all groups are already expanded (making clicking the button a no-op).

Maybe we can take a UI-rework pass over this section of the puzzle list page at some later point and try to file down some of the rough edges here.

Reply:
OK, made a few more changes: I pulled the "expand all" button down into its own row that disappears when you're in unlock mode. I find it less jarring because that section of the UI is already going to churn when you switch modes. I also disabled it when all groups are expanded, and switched it to secondary colors which are a little quieter:

https://user-images.githubusercontent.com/28167/155387495-b185a645-72c2-41aa-83c7-dbb6a0d7c5ab.mov

(I did consider a "collapse all" button for parity, but I think there are both practical and philosophical reasons to not. Practically, it's kind of annoying to tell if any groups are uncollapsed because it's kind of annoying to enumerate all of the groups. Philosophically, I'm not sure it makes sense to look at JR with all groups collapsed. Like, what's the point of having the puzzle list open at all at that point?)

Reply:
I like it!  Thanks for the iteration! :)